components/dashboard/src/data/workspaces/listen-to-workspace-ws-messages.ts

```diff
@@ -51,12 +51,51 @@ export const useListenToWorkspacesWSMessages = () => {
     }, [organizationId, queryClient]);
 };
 
-export function watchWorkspaceStatus(
-    workspaceId: string | undefined,
-    cb: (response: WatchWorkspaceStatusResponse) => void,
-): Disposable {
+type WatchWorkspaceStatusCallback = (response: WatchWorkspaceStatusResponse) => Promise<void> | void;
+
+export function watchWorkspaceStatus(workspaceId: string | undefined, cb: WatchWorkspaceStatusCallback): Disposable {
     return stream<WatchWorkspaceStatusRequest>(
         (options) => workspaceClient.watchWorkspaceStatus({ workspaceId }, options),
         cb,
     );
 }
+
+const cachedCallbackInfoMap = new Map<string, { complete: WatchWorkspaceStatusCallback; priority: number }[]>();
+const cachedDisposables = new Map<string, Disposable>();
+
+// watchWorkspaceStatusInOrder watches the workspace status locally in order of priority.
+export function watchWorkspaceStatusInOrder(
+    workspaceId: string | undefined,
+    priority: number,
+    callback: WatchWorkspaceStatusCallback,
+): Disposable {
+    const wsID = workspaceId || "ALL_WORKSPACES";
+    const newInfo = { complete: callback, priority };
+    const callbacks = cachedCallbackInfoMap.get(wsID) || [];
```

Comment:
nit: on line 92 we're using the nullish coalescing operator, so let's use it here as well for consistency.
```suggestion
    const callbacks = cachedCallbackInfoMap.get(wsID) ?? [];
```